### ISSUE TITLE
Does nvim-lint has a file length limit?

### ISSUE THREAD

**Issue by cpkio** [mfussenegger/nvim-lint#127]
I'm linting my RST with Vale and found out `nvim-lint` does not show diagnostics after ~1000th line. When the problematic text is copied to another file, the diagnostics shows fine. In command-line Vale reports the error in text mode and JSON mode. So why is the diagnostics lost?

I've tried Neovim 6.0 and it's even more buggy (misplaced selections, diagnostics goes to DiagnosticsAll).

**Comment by mfussenegger** [comment]
There is no file length limit induced by nvim-lint.

I tried to reproduce it - but on a 3000 lines file I get errors/warnings near the end with vale without problem.
Can you provide a minimal reproduction?

> I've tried Neovim 6.0 and it's even more buggy (misplaced selections, diagnostics goes to DiagnosticsAll).

What's misplaced selections and what is `DiagnosticsAll` ? Please elaborate.

**Comment by cpkio** [comment]
Sorry, I was thinking `:Diagnostics` was `nvim-lint` command, it's `fzf-lsp.nvim`. Nvim 0.6 behaves erroneously with `nvim-lint`, but let's ignore this for now.

I've made a separate installation with only Treesitter and `nvim-lint` plugins, with 1 line `init.lua`. I've made errors in the document; Vale does catch them in command line. In Neovim, the errors are reported on lines 906 and 939, and no error is reported on lines 966 and 967.

The important thing to note here is my file is Cyrillics mostly, not ASCII. When I replace all the Cyrillics with 'z', everything seems to be worling. Sadly, I cannot provide the text, it's NDA'd. By deleting pieces I found out that the limit goes around 948 lines and 66462 bytes file size.

**Comment by cpkio** [comment]
Aaand by `vim.inspect`ing the `vim.fn.json_decode(output)` from `vale.lua` we find out it IS trimmed somewhere on the way.

**Comment by cpkio** [comment]
You will not believe what did the trick for me (on Windows, if that matters).
https://github.com/mfussenegger/nvim-lint/blob/15e3d372dabdbd52d96d9050d6e0b4e6bbfa9933/lua/lint.lua#L232
should be
`    stdin:write('\n', function()`

**Comment by mfussenegger** [comment]
Could you also try with this change:

```
diff --git a/lua/lint.lua b/lua/lint.lua
index f9fb834..707a6d3 100644
--- a/lua/lint.lua
+++ b/lua/lint.lua
@@ -227,12 +227,11 @@ function M.lint(linter, client_id)
   if linter.stdin then
     local lines = vim.api.nvim_buf_get_lines(0, 0, -1, true)
     for _, line in ipairs(lines) do
       stdin:write(line .. '\n')
     end
-    stdin:write('', function()
-      stdin:shutdown()
+    stdin:shutdown(function()
       stdin:close()
     end)
   else
     stdin:close()
   end
```

Would be interesting to know if that works as well - otherwise I think changing it to write another `\n` should be save without affecting the other linters.

**Comment by cpkio** [comment]
Yes, this does work too.
